xonsh/proc.py

```diff
@@ -528,3 +528,33 @@ def _wcode_to_popen(code):
     else:
         # Can this happen? Let's find out. Returning None is not an option.
         raise ValueError("Invalid os.wait code: {}".format(code))
+
+
+_CCTuple = namedtuple("_CCTuple", ["stdin",
+                                   "stdout",
+                                   "stderr",
+                                   "pid",
+                                   "returncode",
+                                   "args",
+                                   "alias",
+                                   "stdin_redirect",
+                                   "stdout_redirect",
+                                   "stderr_redirect"])
+
+class CompletedCommand(_CCTuple):
+    """
+    Represents a completed subprocess-mode command.
+    """
+
+    def __bool__(self):
+        return self.returncode == 0
+
+    def __eq__(self, other):
+        return self.returncode == other
+
+def make_completed_command(**kwargs):
```

Comment:
I think you can simply add the line:

```python
CompletedCommand.__new__.__defaults__ = (None,) * len(CompletedCommand._fields)
```

This is basically what we do for [`VarDocs`](https://github.com/scopatz/xonsh/blob/9125e0d68dc008e29e300a8ba5d58b4394c37617/xonsh/environ.py#L230). For the reason why this works, see http://stackoverflow.com/a/18348004/2312428